### PR TITLE
[CWS Agent] Runtime policy check cmd shows RC policies too

### PR DESCRIPTION
**Pull request by modernplumbing**
### What does this PR do?
- `sudo ./bin/security-agent/security-agent runtime policy check --loaded-policies` prints the kfilter report of the loaded ruleset with the config values from the running system probe.

- `sudo ./bin/security-agent/security-agent runtime policy check` remains the same, printing the kfilter report of the policies in the policies directory

- Now printing the type of approver in text rather than as the enum int.

- Added a mock of the Runtime Security Client to facilitate testing.

### Describe how to test/QA your changes

Before:
```
vagrant@agent-dev-ubuntu-22:/git/datadog-agent$ ./bin/security-agent/security-agent runtime policy check
{
        "Policies": {
                "exec": {
                        "Mode": "accept",
                        "Flags": [
                                "basename",
                                "flags",
                                "mode"
                        ],
                        "Approvers": null
                },
                "open": {
                        "Mode": "deny",
                        "Flags": [
                                "basename",
                                "flags",
                                "mode"
                        ],
                        "Approvers": {
                                "open.file.path": [
                                        {
                                                "Field": "open.file.path",
                                                "Value": "/etc/shadow",
                                                "Type": 1
                                        },
                                        {
                                                "Field": "open.file.path",
                                                "Value": "/etc/gshadow",
                                                "Type": 1
                                        }
                                ],
                               "open.flags": [
                                        {
                                                "Field": "open.flags",
                                                "Value": 64,
                                                "Type": 1
                                        }
                                ]
                        }
                }
        }
}
```

with this corresponding policy
```
macros:
  - id: NET_UTILS
    description: Executables of common network utilites
    expression: '["socat", "dig", "nslookup", "host", ~"netcat*", ~"nc*", "ncat"]'
rules:
  - id: snapshot
    expression: exec.comm in NET_UTILS
  - id: has_approver
    expression: open.file.path == "/etc/gshadow" || open.file.path == "/etc/shadow"
  - id: has_approver_2
    expression: open.flags & (O_CREAT | O_EXCL) == O_CREAT
```
After:
```
vagrant@agent-dev-ubuntu-22:/git/datadog-agent$ sudo ./bin/security-agent/security-agent runtime policy check --loaded-policies
{
        "Policies": {
                "exec": {
                        "Mode": "accept",
                        "Flags": [
                                "basename",
                                "flags",
                                "mode"
                        ],
                        "Approvers": null
                },
                "open": {
                        "Mode": "deny",
                        "Flags": [
                                "basename",
                                "flags",
                                "mode"
                        ],
                        "Approvers": {
                                "open.file.path": [
                                        {
                                                "Field": "open.file.path",
                                                "Value": "/etc/gshadow",
                                                "Type": "scalar"
                                        },
                                        {
                                                "Field": "open.file.path",
                                                "Value": "/etc/shadow",
                                                "Type": "scalar"
                                        }
                                ],
                                "open.flags": [
                                        {
                                                "Field": "open.flags",
                                                "Value": 64,
                                                "Type": "scalar"
                                        }
                                ]
                        }
                }
... [Plus content from Remote Config]
        }
}
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/agent/client_mocks.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+package agent
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/security/proto/api"
+	"math"
+)
+
+// Mocks
+
+type MockRuntimeSecurityClient struct{}
```
Comment (paulcacheux) [inline pkg/security/agent/client_mocks.go:15]: Is this generated manually ? Why not using the one generated with `inv -e security-agent.gen-mocks` ?

File: pkg/security/probe/kfilters/policy.go [hunk 2]
```diff
@@ -59,15 +59,26 @@ func (m PolicyMode) MarshalJSON() ([]byte, error) {
 
 // MarshalJSON returns the JSON encoding of the policy flags
 func (f PolicyFlag) MarshalJSON() ([]byte, error) {
+	flags := f.StringArray()
+
+	for idx, flag := range flags {
+		flags[idx] = "\"" + flag + "\""
+	}
+
+	return []byte("[" + strings.Join(flags, ",") + "]"), nil
```
Comment (paulcacheux) [inline pkg/security/probe/kfilters/policy.go:68]: why not using `json.Marshal` directly ? is this performance sensitive ?
https://go.dev/play/p/BxQAehD1wj5

File: pkg/security/proto/api/README.md [hunk 3]
```diff
@@ -11,5 +11,5 @@ to install the correct version of required tools
 
 From the repository root run the following:
 ```
-inv -e generate-cws-proto
+inv -e security-agent.generate-cws-proto
```
Comment (paulcacheux) [inline pkg/security/proto/api/README.md:14]: Thanks !

File: pkg/security/proto/api/transform_functions.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+package api
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/security/probe/kfilters"
+	"github.com/DataDog/datadog-agent/pkg/security/secl/compiler/eval"
+	"github.com/DataDog/datadog-agent/pkg/security/secl/rules"
+	"strconv"
+)
+
+func (protoRuleSetReport *RuleSetReportMessage) FromProtoToKFiltersRuleSetReport() *kfilters.ApplyRuleSetReport {
```
Comment (paulcacheux) [inline pkg/security/proto/api/transform_functions.go:17]: I know you were not super happy with those functions, but I feel like it's ok in the end

File: pkg/security/agent/client_mocks.go (hunk 1 above)
Comment (modernplumbing) [reply pkg/security/agent/client_mocks.go:15]: The Runtime Security Client doesn't fit the interface defined in api.SecurityModuleClient (on which the mocks generated by `inv -e security-agent.gen-mocks` are based on). Which is why I created another interface `SecurityModuleClientWrapper`. https://github.com/DataDog/datadog-agent/blob/9e867fd144a1582260014bb939f7dcf5ad35639b/pkg/security/agent/client.go#L21

File: pkg/security/probe/kfilters/policy.go (hunk 2 above)
Comment (modernplumbing) [reply pkg/security/probe/kfilters/policy.go]: good catch!

File: pkg/security/agent/client_mocks.go (hunk 1 above)
Comment (paulcacheux) [reply pkg/security/agent/client_mocks.go:15]: ah yeah alright

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**